janus_server/src/aggregator.rs

```diff
@@ -101,26 +107,129 @@ impl From<datastore::Error> for Error {
     }
 }
 
-/// A PPM aggregator.
-// TODO: refactor Aggregator to be non-task-specific (look up task-specific data based on task ID)
-// TODO: refactor Aggregator to perform indepedent batched operations (e.g. report handling in
-//       Aggregate requests) using a parallelized library like Rayon.
+/// Aggregator implements a PPM aggregator.
 pub struct Aggregator<C: Clock> {
-    /// The task being aggregated.
-    task: Task,
-    /// The datastore used for durable storage.
+    /// Datastore used for durable storage.
     datastore: Arc<Datastore>,
-    /// The clock to use to sample time.
+    /// Clock used to sample time.
     clock: C,
+    /// Cache of task aggregators.
+    task_aggregators: Mutex<HashMap<TaskId, Arc<TaskAggregator>>>,
+}
+
+impl<C: Clock> Aggregator<C> {
+    fn new(datastore: Arc<Datastore>, clock: C) -> Self {
+        Self {
+            datastore,
+            clock,
+            task_aggregators: Mutex::new(HashMap::new()),
+        }
+    }
+
+    async fn handle_hpke_config(&self, task_id_base64: &[u8]) -> Result<Vec<u8>, Error> {
+        let task_id_bytes = base64::decode_config(task_id_base64, base64::URL_SAFE_NO_PAD);
+        if task_id_bytes.is_err() {
+            return Err(Error::UnrecognizedMessage("task_id", None));
+        }
+        let task_id = TaskId::get_decoded(&task_id_bytes.unwrap());
+        if task_id.is_err() {
+            return Err(Error::UnrecognizedMessage("task_id", None));
+        }
```

Comment:
I think this would be more idiomatic:
```
let task_id_bytes = base64::decode_config(...)
    .map_err(|_| Error::UnrecognizedMessage("task_id", None))?;
...
let task_id = TaskId::get_decoded(&task_id_bytes)
    .map_err(|_| Error::UnrecognizedMessage("task_id", None))?;
```